Add isItemValid check for ItemStackHandler

RedBorg opened the pull request:
ItemStackHandler will now no longer proceed to insert an item if isItemValid returns false.

This fixes #5849.

This handler will now no longer accept anything other than diamonds:
```java
ItemStackHandler handler = new ItemStackHandler(5) {
    @Override
    public boolean isItemValid(int slot, ItemStack stack) {
        return stack.getItem() == Items.DIAMOND;
    }
}
```

RedBorg committed:
Add isItemValid check for ItemStackHandler

ItemStackHandler will now no longer proceed to insert an item if isItemValid returns false.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/MinecraftForge/MinecraftForge?pullRequest=5850) <br/>All committers have signed the CLA.

loordgek commented:
dont put it there it will only check when the existing is not empty
put it here https://github.com/MinecraftForge/MinecraftForge/blob/5fd0e5c8cff2abf60036eb004ad58e9a7752da1d/src/main/java/net/minecraftforge/items/ItemStackHandler.java#L83

RedBorg commented:
@loordgek 

Oops, my bad! Shouldn’t have gone that fast, thanks for the heads-up.

Going to fix that.

RedBorg committed:
Fixed logic error in the isItemValid patch

The isItemValid check in ItemStackHandler wouldn’t be made if the slot was empty, fixed that.
